Documentation/signing-and-verification-guide.md

```diff
@@ -0,0 +1,265 @@
+* hello-0.0.1-linux-amd64.aci
+* pubkeys.gpg
+
+### Rocket Integration
+
+Lets walk through the steps rocket takes when installing images using Meta Discovery.
+The following rocket command:
+
+```
+$ rkt run example.com/hello:0.0.1
+```
+
+translates into the following curl commands:
+
+```
+$ curl https://example.com/hello?ac-discovery=1
+$ curl https://example.com/images/example.com/hello-0.0.1-linux-amd64.aci
+$ curl https://example.com/images/example.com/hello-0.0.1-linux-amd64.sig
+```
+
+The first response contains the template URL used to download the ACI image and detached signature file.
+
+```
+<meta name="ac-discovery" content="example.com/hello https://example.com/images/{name}-{version}-{os}-{arch}.{ext}">
+```
+
+Rocket populates the `{os}` and `{arch}` based on the current running system.
+The `{version}` will be taken from the tag given on the command line.
+The `{ext}` is based on convention and .aci will be used for ACI images and .sig will be used for detached signatures.
+
+Once the ACI image has been downloaded rocket will extract the ACName from the image metadata. The ACName will be used to locate trusted public keys in the rocket keystore and perform signature validation.
+
+## Verifying Images with Rocket
+
+### Establishing Trust
+
+By default rocket does not trust any signing keys. Trust is established by storing public keys in the rocket keystore.
+The following directories make up the default rocket keystore layout:
+
+```
+/etc/rkt/trustedkeys/root.d
+/etc/rkt/trustedkeys/prefix.d
+/usr/lib/rkt/trustedkeys/root.d
+/usr/lib/rkt/trustedkeys/prefix.d
+```
+
+Trusted keys are saved in the desired directory named after the fingerprint of the public key.
+
+### Trusting the example.com/hello key
+
+#### Download the public key
+
+```
+$ curl -O https://example.com/pubkeys.gpg
+```
+
+#### Capture the public key fingerprint
+
+```
+$ gpg2 --no-default-keyring --fingerprint --keyring ./pubkeys.gpg kelsey.hightower@coreos.com
+```
+
+```
+pub   2048R/26EF7A14 2015-01-09
+      Key fingerprint = B346 E31D E7E3 C6F9 D1D4  603F 4DFB 61BF 26EF 7A14
+uid       [ unknown] Kelsey Hightower (ACI signing key) <kelsey.hightower@coreos.com>
+sub   2048R/B9C074CD 2015-01-09
+```
+
+Remove white spaces and convert to lowercase:
+
+```
+$ echo "B346 E31D E7E3 C6F9 D1D4  603F 4DFB 61BF 26EF 7A14" | \
+tr -d "[:space:]" | tr '[:upper:][:lower:]' '[:lower:][:upper:]'
+```
+```
+b346e31de7e3c6f9d1d4603f4dfb61bf26ef7a14
+```
+
+#### Trust the key for the example.com/hello prefix
+
+```
+mkdir -p /etc/rkt/trustedkeys/prefix.d/example.com/hello
+mv pubkeys.gpg  /etc/rkt/trustedkeys/prefix.d/example.com/hello/b346e31de7e3c6f9d1d4603f4dfb61bf26ef7a14
+```
+
+Now the public key with fingerprint `b346e31de7e3c6f9d1d4603f4dfb61bf26ef7a14` will be trusted for all images with a name prefix of `example.com/hello`.
+If you would like to trust a public key for any image store the public key in one of the following directories:
```

Comment:
Would be nice if we called out the difference between /usr and /etc and I think we should only encourage people to use the latter.